keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/NotificationEventListener.kt

```diff
@@ -240,6 +243,46 @@ class NotificationEventListener(
     }
   }
 
+  @EventListener(ConstraintStateChanged::class)
+  fun onConstraintStateChanged(notification: ConstraintStateChanged) {
+    log.debug("Received constraint state changed event: $notification")
+    with(notification) {
+      // if this is the first time the constraint was evaluated, send a notification
+      // so the user can react via other interfaces outside the UI (e.g. e-mail, Slack)
+      if (constraint is ManualJudgementConstraint &&
+        previousState == null &&
+        currentState.status == ConstraintStatus.PENDING
+      ) {
+        val config = repository.getDeliveryConfig(currentState.deliveryConfigName)
+
+        val deliveryArtifact = config.artifacts.find {
+          it.reference == currentState.artifactReference
+        } ?: return
```

Comment:
nit: Add a logging statement here if we can't find the artifact.
```suggestion
        }.also { 
          if (it == null) log.debug("Artifact with reference ${currentState.artifactReference}  not found in delivery config" 
        } ?: return
```